src/main/java/com/letsparty/web/controller/PartyController.java

```diff
@@ -224,6 +225,42 @@ public String withdraw(@PathVariable("partyNo") int partyNo, @PathVariable("upaN
 		return "redirect:/party/{partyNo}";
 	}
 	
+	// 퇴장시키기 화면
+	@PreAuthorize("isAuthenticated()")
+	@GetMapping("/{partyNo}/setting/member")
+	public String kickOutPage(@PathVariable int partyNo, Model model, @AuthenticationPrincipal LoginUser loginUser) {
+		List<UserPartyApplication> userPartyApplications = partyService.getUserPartyApplications(partyNo, loginUser.getNo());
+		Party savedParty = partyService.getPartyByNo(partyNo);
+		// 자신이 해당 파티의 리더가 아니라면 리다이렉트
+		if(!savedParty.getLeader().getId().equals(loginUser.getId())) {
+			return "redirect:/party/{partyNo}";
+		}
+		model.addAttribute("users", userPartyApplications);
+		model.addAttribute("partyNo", partyNo);
+		model.addAttribute("profilesPath", profilesPath);
+		return "page/party/kick-out";
+	}
+	
+	// 퇴장시키기
+	@PostMapping("/{partyNo}/setting/kick")
+	@ResponseBody
+	public ResponseEntity<Map<String, Object>> kickOut(@PathVariable int partyNo,
+			@RequestParam String userId /* @AuthenticationPrincipal LoginUser loginUser */) {
+		Map<String, Object> response = new HashMap<>();
+	    UserPartyApplication savedUserPartyApplication = userPartyApplicationService.findByPartyNoAndUserId(partyNo, userId);
+	    if(savedUserPartyApplication == null) {
+	        response.put("status", "error");
+	        response.put("message", "유저를 찾을 수 없습니다.");
+	        return ResponseEntity.badRequest().body(response);
+	    }
+	    //TODO 멤버 강퇴를 시도하는 유저가 리더인지, 자기 자신을 탈퇴하려고 시도하지 않는지 loginUser인자를 사용하여 추가로 검사할 수 있음
+	    
+	    userPartyApplicationService.update(savedUserPartyApplication);
+	    response.put("status", "success");
+	    response.put("message", "퇴장 처리가 완료되었습니다.");
+	    return ResponseEntity.ok(response);
+	}
```

Comment:
```suggestion
	// 퇴장시키기
	@PostMapping("/{partyNo}/setting/kick")
	@ResponseBody
	public ResponseEntity<Map<String, Object>> kickOut(@PathVariable int partyNo, @RequestParam String userId, @AuthenticationPrincipal LoginUser loginUser) {
		Map<String, Object> response = validationService.kickOutUser(partyNo, userId, loginUser.getId());
	    if ("error".equals(response.get("status"))) {
	        return ResponseEntity.badRequest().body(response);
	    }
	    return ResponseEntity.ok(response);
	}
```
컨트롤러에서 이렇게 작성하고 ValidationService에서 kickOutUser메소드를 만들어서 그 안에서 response응답을 보내주려는데 어떨까요?

Reply:
``` 
public Map<String, Object> kickOutUser(int partyNo, String userId, String loginUser) {
        Map<String, Object> response = new HashMap<>();
        UserPartyApplication savedUserPartyApplication = userPartyApplicationService.findByPartyNoAndUserId(partyNo, userId);
        Party savedParty = partyService.getPartyByNo(partyNo);

        if (savedUserPartyApplication == null) {
            response.put("status", "error");
            response.put("message", "유저를 찾을 수 없습니다.");
            return response;
        }
        if (!savedParty.getLeader().getId().equals(loginUser)) {
            response.put("status", "error");
            response.put("message", "리더만 멤버를 탈퇴 시킬 수 있습니다.");
            return response;
        }
        if (savedParty.getLeader().getId().equals(userId)) {
            response.put("status", "error");
            response.put("message", "리더는 자기 자신을 탈퇴 시킬 수 없습니다.");
            return response;
        }
        userPartyApplicationService.update(savedUserPartyApplication);
        response.put("status", "success");
        response.put("message", "퇴장 처리가 완료되었습니다.");
        return response;
	}
```
이건 kickOutUser 메소드 코드입니다.